test/jdk/javax/swing/JComboBox/EditableComboBoxPopupPos.java

```diff
@@ -0,0 +1,139 @@
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+/*
+ * @test
+ * @key headful
+ * @bug 8302558
+ * @summary Tests if the Popup from an editable ComboBox with a border
+ *          is in the correct position
+ * @run main EditableComboBoxPopupPos
+ */
+
+import java.awt.Robot;
+import java.awt.event.InputEvent;
+import java.lang.reflect.InvocationTargetException;
+
+import javax.swing.BorderFactory;
+import javax.swing.JComboBox;
+import javax.swing.JFrame;
+import javax.swing.JPanel;
+import javax.swing.SwingUtilities;
+import javax.swing.UIManager;
+import javax.swing.UnsupportedLookAndFeelException;
+
+public class EditableComboBoxPopupPos {
+    private static Robot robot;
+    private static JFrame frame;
+    private static JPanel panel;
+    private static JComboBox cb1, cb2;
+    private static String lafName;
+
+    private static final int BUTTON_OFFSET = 10;
+    private static final int POPUP_OFFSET = 5;
+
+    public static void main(String[] args) throws InterruptedException, InvocationTargetException {
+        try {
+            robot = new Robot();
+            robot.setAutoDelay(100);
+            robot.setAutoWaitForIdle(true);
+
+            for (UIManager.LookAndFeelInfo laf : UIManager.getInstalledLookAndFeels()) {
+                lafName = laf.getName().equals("CDE/Motif") ? "Motif" : laf.getName();
+                SwingUtilities.invokeAndWait(() -> {
+                    setLookAndFeel(laf);
+                    panel = new JPanel();
+                    String[] comboStrings = {"One", "Two", "Three"};
+                    cb1 = new JComboBox(comboStrings);
+                    cb1.setEditable(true);
+                    cb1.setBorder(BorderFactory.createTitledBorder(
+                            "Editable JComboBox"));
+
+                    cb2 = new JComboBox(comboStrings);
+                    cb2.setEditable(true);
+
+                    panel.add(cb1);
+                    panel.add(cb2);
+
+                    frame = new JFrame();
+                    frame.setDefaultCloseOperation(JFrame.EXIT_ON_CLOSE);
+                    frame.add(panel);
+                    frame.pack();
+                    frame.setLocationRelativeTo(null);
+                    frame.setVisible(true);
+                });
+
+                // Change starting selection to check if the position of the
+                // first selection item is in the correct position on screen.
+                cb1.setSelectedIndex(1);
+                cb2.setSelectedIndex(1);
+
+                runTestOnComboBox(cb1);
+                runTestOnComboBox(cb2);
+
+                checkSelection(cb1, cb2);
+            }
+        } catch (Exception e) {
+            e.printStackTrace();
```

Comment:
You should not catch the exception but propagate it for jtreg to fail the test if it throws the exception. Otherwise, the test is considered passed even though it may print `Test failed`.

Reply:
Good point. I propagated the exception and re-tested